Content.Client/Preferences/UI/HumanoidProfileEditor.xaml.cs

```diff
@@ -1411,5 +1558,106 @@ private void OnCheckBoxToggled(BaseButton.ButtonToggledEventArgs args)
+                {
+                    tooltip += "Whitelist:";
+                    if (loadout.EntityWhitelist?.Components != null)
+                        foreach (var require in loadout.EntityWhitelist.Components)
+                            tooltip += $"\n - {require} (Component)";
+                    if (loadout.EntityWhitelist?.Tags != null)
+                        foreach (var require in loadout.EntityWhitelist.Tags)
+                            tooltip += $"\n - {require} (Tag)";
+                    if (loadout.EntityWhitelist?.RequireAll == true)
+                        tooltip += $"\n Require All: {loadout.EntityWhitelist.RequireAll}"; // This comes first because job whitelist has no effect on requireall
+                    if (loadout.JobWhitelist != null)
+                        foreach (var require in loadout.JobWhitelist)
+                            tooltip += $"\n - {require} (Job)";
+                    if (loadout.SpeciesWhitelist != null)
+                        foreach (var require in loadout.SpeciesWhitelist)
+                            tooltip += $"\n - {require} (Species)";
+                }
+                if (loadout.EntityBlacklist != null || loadout.JobBlacklist != null)
+                {
+                    tooltip += "Blacklist:";
+                    if (loadout.EntityBlacklist?.Components != null)
+                        foreach (var require in loadout.EntityBlacklist.Components)
+                            tooltip += $"\n - {require} (Component)";
+                    if (loadout.EntityBlacklist?.Tags != null)
+                        foreach (var require in loadout.EntityBlacklist.Tags)
+                            tooltip += $"\n - {require} (Tag)";
+                    if (loadout.EntityBlacklist?.RequireAll == true)
+                        tooltip += $"\n Require All: {loadout.EntityBlacklist.RequireAll}"; // This comes first because job whitelist has no effect on requireall
+                    if (loadout.JobBlacklist != null)
+                        foreach (var require in loadout.JobBlacklist)
+                            tooltip += $"\n - {require} (Job)";
+                    if (loadout.SpeciesBlacklist != null)
+                        foreach (var require in loadout.SpeciesBlacklist)
+                            tooltip += $"\n - {require} (Species)";
+                }
+
+                if (tooltip != "")
+                {
+                    _checkBox.ToolTip = tooltip;
+                    _checkBox.TooltipDelay = 0.2f;
+                }
+
+                AddChild(new BoxContainer
+                {
+                    Orientation = LayoutOrientation.Horizontal,
+                    Children = { previewLoadout, _checkBox },
+                });
+            }
+
+            private void OnCheckBoxToggled(BaseButton.ButtonToggledEventArgs args)
+            {
+                PreferenceChanged?.Invoke(Preference);
```

Comment:
A) why is there 0 comments for this
B) none of this is localised
C) why is the formatting like this
D) Why didn't you just make a method instead of copy-pasting each whitelist 3 times over.